lib/archethic/mining/pending_transaction_validation.ex

```diff
@@ -70,6 +79,115 @@ defmodule Archethic.Mining.PendingTransactionValidation do
     end
   end
 
+  def validate_previous_transaction_type?(tx) do
+    case Transaction.network_type?(tx.type) do
+      false ->
+        # not a network tx, no need to validate with last tx
+        :ok
+
+      true ->
+        # when network tx, check with previous transaction
+        if validate_network_chain?(tx.type, tx), do: :ok, else: {:error, :invalid_tx_type}
+    end
+  end
+
+  @spec validate_network_chain?(
+          :code_approval
+          | :code_proposal
+          | :mint_rewards
+          | :node
+          | :node_rewards
+          | :node_shared_secrets
+          | :oracle
+          | :oracle_summary
+          | :origin,
+          Archethic.TransactionChain.Transaction.t()
+        ) :: boolean
+  def validate_network_chain?(type, tx = %Transaction{})
+      when type in [:oracle, :oracle_summary] do
+    # mulitpe txn chain based on summary date
+
+    case OracleChain.get_gen_addr() do
+      nil ->
+        false
+
+      gen_addr ->
+        # first adddress found by looking up in the db
+        first_addr =
+          tx
+          |> Transaction.previous_address()
+          |> TransactionChain.get_genesis_address()
+
+        # 1: tx gen & validated in current summary cycle, gen_addr.current must match
+        # 2: tx gen in prev summary cycle &  validated in current summary cycle, gen_addr.prev must match
+        # situation: the shift b/w summary A to summary B
+        gen_addr.current |> elem(0) == first_addr ||
+          gen_addr.prev |> elem(0) == first_addr
+    end
+  end
+
+  def validate_network_chain?(type, tx = %Transaction{})
+      when type in [:mint_rewards, :node_rewards] do
+    # singleton tx chain in network lifespan
+    case Reward.get_gen_addr() do
+      nil ->
+        false
+
+      gen_addr ->
+        # first addr located in db by prev_address from tx
+        first_addr =
+          tx
+          |> Transaction.previous_address()
+          |> TransactionChain.get_genesis_address()
+
+        gen_addr == first_addr
+    end
+  end
+
+  def validate_network_chain?(type, _tx = %Transaction{})
```

Comment:
I don't think you need this one as well as L:186.
You should use a default one to be by-passed